common/versioning/version_tasks.gradle

```diff
@@ -1,17 +1,41 @@
 def getVersionFile() {
-    
-    File file = new File("$rootDir/common/common/versioning/version.properties")
-    if(file.exists()){
-        return file;
-    }else{
-        return new File("$rootDir/common/versioning/version.properties")
-    }
-
+    // Previously we were referencing the versions file using the path from rootDir of the root
+    // project, however, now we've introduced more root projects and using the root dir is no longer
+    // feasible. Therefore, using a project relative path here. Project relative path will resolve
+    // the file path relative to the root of the project that is using the path (as opposed to the
+    // root of the root project from where the build was triggered).
+    //
+    // This version file is located in the common module in the common repo i.e it is located in
+    // {repo_root_dir}/common
+    //
+    // This same version file is being used by four modules:
+    // - common, keyvault, labapi, testutils, uiautomationutilities
+    //
+    // When using the project (module) reference for the version file, gradle will try to resolve
+    // the path based on the module that was being built. For instance, when building common it
+    // would resolve it as "{repo_root_dir}/common/versioning/version.properties"
+    //
+    // However, when building keyvault it would resolve it as
+    // "{repo_root_dir}/keyvault/versioning/version.properties"
+    // but that path is wrong...because the file is located at
+    // "{repo_root_dir}/common/versioning/version.properties"
+    // and as said above, the same file from common module is used across common, keyvault, labapi,
+    // testutils, uiautomationutilities
+    // This was probably done to version the LATEST (0.0.+) versions with the same logic.
+    // So to get the right path here, we're goind to do a String replace and replace the module name
+    // occuring in the path, always with common because that's where the version file is located.
+    String path = file("versioning/version.properties").getAbsolutePath()
```

Comment:
as discussed, let's move the file up one level instead of doing this workaround.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1468/commits/8eecd28712ba124d1996a5d68f2dbe6e6b880632